STYLE.md

```diff
@@ -0,0 +1,186 @@
+them.
+
+C++ style line comments should be preferred over C-style block comments.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+  
+```go
+/*
+  Unmarshal converts a JSON string into a go interface
+  ...
+*/
+```
+  
+</td><td>
+
+```go
+// Unmarshal converts a JSON string into a go interface
+// ...
+```
+ 
+</td></tr>
+</tbody></table>
+
+### Package Comments
+
+Package comments should start with the word "Package" followed by the package name. For packages
+with multiple files or with a large documentation, use a separate `doc.go` file for the  comment.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+  
+```go
+// sort is a package which implements sorting functions.
+```
+  
+</td><td>
+
+```go
+// Package sort implements sorting functions.
+```
+ 
+</td></tr>
+</tbody></table>
+
+### Documentation Comments
+
+All doc comments for symbols should start with the symbol name.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+  
+```go
+// Function Quick is an implementation
+// of the Quicksort algorithm.
+```
+  
+</td><td>
+
+```go
+// Quick is an implementation
+// of the Quicksort algorithm.
+```
+ 
+</td></tr>
+</tbody></table>
+
+### Author Comments
+
+A comment recording the author of a particular file may also be added. This comment should be
+written at the top of the file and it should not be a part of the package documentation.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+  
+```go
+// author: Rak Laptudirm (@raklaptudirm)
+package sort
+```
+Comment is a part of the package documentation.
+  
+</td><td>
+
+```go
+// author: Rak Laptudirm (@raklaptudirm)
+
+package sort
+```
+Comment is not a part of the package documentation.
+ 
+</td></tr>
+</tbody></table>
+
+## Naming
+
+### Package Naming
+
+Package names should be short and to the point. Keep in mind that this identifier will be used to
+refer to your package, so make it easy to remember. It should be only composed of lower case
+letters. Prefer `json` to `JSON` or `Json`. If your package name has two words, merge them
+together. Prefer `jsonencodeing` to `jsonEncoding` or `json_encodeing`.
+
+Add the `_test` suffix to your package name to implement black-box testing for your package.
+
+### Symbol Naming
+
+Go symbols should be named in the `camelCase` or `UpperCamelCase`, depending of whether the symbol
+is exported or not. The case when using acronymns in names should be consistent. Use `json` or
+`JSON` instead of `Json`.
+
+For exported symbols, use the package name to your advantage to write concise symbol names. For
+example, if you have a package `png`, instead of defining a `png.PNGFile`, define a `png.File`.
+It is much clearer and avoids repetition.
+
+### Function Naming
+
+#### Constructor Naming
+
+Constructors should use the naming format `New<Type>()`. If the package only exports a single
```

Comment:
This is somewhat relevant, `New<Something>` should return a pointer to be consistent with `new` in built function and `Make<Something>` should return an object to be consistent with `make` in built function.